libraries/AP_ESC_Telem/AP_ESC_Telem.cpp

```diff
@@ -305,6 +305,44 @@ bool AP_ESC_Telem::get_usage_seconds(uint8_t esc_index, uint32_t& usage_s) const
     return true;
 }
 
+#if AP_EXTENDED_ESC_TELEM_ENABLE
+// get an individual ESC's input duty if available, returns true on success
+bool AP_ESC_Telem::get_input_duty(uint8_t esc_index, uint8_t& input_duty) const
+{
+    if (esc_index >= ESC_TELEM_MAX_ESCS
+        || AP_HAL::millis() - _telem_data[esc_index].last_update_ms > ESC_TELEM_DATA_TIMEOUT_MS
```

Comment:
i think adding a bool function data_stale which takes a telem type, and does this logic in one place would be worthwhile

Reply:
replaced with telemdata.stale()